src/controllers/aer_controller.hpp

```diff
@@ -418,49 +415,39 @@ void Controller::set_config(const Config &config) {
 #endif
 
   // Load configurations for parallelization
-
-  if (config.max_memory_mb.has_value())
-    max_memory_mb_ = config.max_memory_mb.value();
+  config.get_option(max_memory_mb_, "max_memory_mb");
 
   // for debugging
-  if (config._parallel_experiments.has_value()) {
-    parallel_experiments_ = config._parallel_experiments.value();
+  if (config.get_option(parallel_experiments_, "_parallel_experiments"))
     explicit_parallelization_ = true;
-  }
 
   // for debugging
-  if (config._parallel_shots.has_value()) {
-    parallel_shots_ = config._parallel_shots.value();
+  if (config.get_option(parallel_shots_, "_parallel_shots"))
     explicit_parallelization_ = true;
-  }
 
   // for debugging
-  if (config._parallel_state_update.has_value()) {
-    parallel_state_update_ = config._parallel_state_update.value();
+  if (config.get_option(parallel_state_update_, "_parallel_state_update"))
     explicit_parallelization_ = true;
-  }
 
   if (explicit_parallelization_) {
     parallel_experiments_ = std::max<int>({parallel_experiments_, 1});
     parallel_shots_ = std::max<int>({parallel_shots_, 1});
     parallel_state_update_ = std::max<int>({parallel_state_update_, 1});
   }
 
-  if (config.accept_distributed_results.has_value())
-    accept_distributed_results_ = config.accept_distributed_results.value();
+  config.get_option(accept_distributed_results_, "accept_distributed_results");
 
   // enable multiple qregs if cache blocking is enabled
-  if (config.blocking_qubits.has_value())
-    cache_block_qubit_ = config.blocking_qubits.value();
+  config.get_option(cache_block_qubit_, "blocking_qubits");
 
   // enable batched multi-shots/experiments optimization
-  batched_shots_gpu_ = config.batched_shots_gpu;
-  batched_shots_gpu_max_qubits_ = config.batched_shots_gpu_max_qubits;
+  config.get_option(batched_shots_gpu_, "batched_shots_gpu");
+  config.get_option(batched_shots_gpu_max_qubits_,
+                    "batched_shots_gpu_max_qubits");
```

Comment:
This doesn't necessarily need changing in this PR, but more just a comment: the top message suggests that the infrequently used options are just stored in the `options`, not directly looped through in Python space to bind.

If that's the case, how much time are we spending now binding these options to the instance members in C++ space?  If they're unlikely to actually be accessed, perhaps we should consider just storing the `config` as a member on the class, and turning all the individual options members into getter methods that do the parsing?

It would have the downside of configuration errors getting raised later in the work flow, potentially, but if the binding is a performance bottleneck, that might be worth it?

Reply:
Though I need to get profile data again, CPU was used to copy `AER::Config` in the last experiment. This is because fields of its class are many.